<launch>
  <arg name="robot_name"/>

  <arg name="robot1_init_x" default="0.0"/>
  <arg name="robot1_init_y" default="0.0"/>
  <arg name="robot1_init_yaw" default="0.0"/>

  <arg name="robot2_init_x" default="1.5"/>
  <arg name="robot2_init_y" default="0.0"/>
  <arg name="robot2_init_yaw" default="0.0"/>

  <group ns="$(arg robot_name)">
    <include if="$(eval arg('robot_name') == 'robot1')" file="$(find map_merge)/launch/includes/map_merge_tf_publisher.launch.xml">
      <arg name="robot_name" value="$(arg robot_name)"/>
      <arg name="init_x" value="$(arg robot1_init_x)"/>
      <arg name="init_y" value="$(arg robot1_init_y)"/>
      <arg name="init_yaw" value="$(arg robot1_init_yaw)"/>
    </include>

    <include if="$(eval arg('robot_name') == 'robot2')" file="$(find map_merge)/launch/includes/map_merge_tf_publisher.launch.xml">
      <arg name="robot_name" value="$(arg robot_name)"/>
      <arg name="init_x" value="$(arg robot2_init_x)"/>
      <arg name="init_y" value="$(arg robot2_init_y)"/>
      <arg name="init_yaw" value="$(arg robot2_init_yaw)"/>
    </include>

    <include if="$(eval arg('robot_name') == 'robot1')" file="$(find map_merge)/launch/includes/map_merge.launch.xml">
      <arg name="robot_name" value="$(arg robot_name)"/>

      <arg name="robot1_init_x" value="$(eval arg ('robot1_init_x') / -0.05)"/>
      <arg name="robot1_init_y" value="$(eval arg ('robot1_init_y') / -0.05)"/>
      <arg name="robot1_init_yaw" value="$(arg robot1_init_yaw)"/>

      <arg name="robot2_init_x" value="$(eval arg ('robot2_init_x') / -0.05)"/>
      <arg name="robot2_init_y" value="$(eval arg ('robot2_init_y') / -0.05)"/>
      <arg name="robot2_init_yaw" value="$(arg robot2_init_yaw)"/>

      <arg name="map_num" value="1"/>

    </include>

    <include if="$(eval arg('robot_name') == 'robot2')" file="$(find map_merge)/launch/includes/map_merge.launch.xml">
      <arg name="robot_name" value="$(arg robot_name)"/>

      <arg name="robot1_init_x" value="$(eval arg ('robot1_init_x') / -0.05)"/>
      <arg name="robot1_init_y" value="$(eval arg ('robot1_init_y') / -0.05)"/>
      <arg name="robot1_init_yaw" value="$(arg robot1_init_yaw)"/>

      <arg name="robot2_init_x" value="$(eval arg ('robot2_init_x') / -0.05)"/>
      <arg name="robot2_init_y" value="$(eval arg ('robot2_init_y') / -0.05)"/>
      <arg name="robot2_init_yaw" value="$(arg robot2_init_yaw)"/>

      <arg name="map_num" value="2"/>

    </include>

    <node pkg="robot_pose_publisher" type="robot_pose_publisher" name="map_merge_odom_publisher">
      <param name="map_frame" value="$(arg robot_name)/merge_map"/>
      <param name="base_frame" value="$(arg robot_name)/base_link"/>
      <param name="is_stamped" value="true"/>
    </node>
  </group>
</launch>
